### PR TITLE
Set onnx opset version before model select

### PR DESCRIPTION
**Pull request by suffiank**
Set opset version before model select call - which is used to trigger warnings.

### CODE REVIEW

**Comment by spandantiwari** [comment]
@suffiank - can you elaborate in the description what issue are you trying to solve?

File: torch/onnx/utils.py [hunk 1]
```diff
@@ -498,9 +498,9 @@ def _export(model, args, f, export_params=True, verbose=False, training=None,
         # If you really know what you're doing, you can turn
         # training=TrainingMode.TRAINING or training=TrainingMode.PRESERVE,
         # (to preserve whatever the original training mode was.)
+        _set_opset_version(opset_version)
+        _set_operator_export_type(operator_export_type)
```
Comment (spandantiwari) [inline torch/onnx/utils.py:502]: Will this set the original mode of the model (eval vs train) after the export? @lara-hdr

File: torch/onnx/utils.py [hunk 2]
```diff
@@ -43,8 +43,8 @@ def select_model_mode_for_export(model, mode):
             if is_originally_training:
                 warnings.warn("You are exporting the model to ONNX while in training mode with "
                               "'train' parameter not specified. The model will default to inference mode export. "
-                              "If you wish to export a training amenable ONNX model, specify train=TrainingMode.TRAIN or "
-                              "train=TrainingMode.PRESERVE (to preserve the original model state) in torch.onnx.export().")
+                              "If you wish to export a training amenable ONNX model, specify training=TrainingMode.TRAINING or "
```
Comment (spandantiwari) [inline torch/onnx/utils.py:46]: This looks good.

**Comment by spandantiwari** [comment]
@suffiank - can you please add a test to cover this?

File: torch/onnx/utils.py (hunk 1 above)
Comment (suffiank) [reply torch/onnx/utils.py:502]: So I noticed the message print even when I set opset 12. Initially I thought the parameter wasn't being respected and an edit was required. Turns out it's just a bad message. I then matched the call sequence as in the pretty export function.

Comment (suffiank) [reply torch/onnx/utils.py:502]: So I noticed the message print even when I set opset 12. Initially I thought the parameter wasn't being respected and an edit was required. Turns out it's just a bad message. I then matched the call sequence as in the pretty export function.

**Comment by suffiank** [comment]
> @suffiank - can you please add a test to cover this?

So I already see tests to cover expected opset version. And also model state is preserved after export (eval vs. training). Since this only fixes printing of warning message, it's better without?

**Comment by lara-hdr** [comment]
Thanks @suffiank. I had also created a similar PR https://github.com/pytorch/pytorch/pull/37397.
@houseroad could you please review/merge either PR?

**Review by lara-hdr — APPROVED** [review]

**Review by houseroad — APPROVED** [review]

**Pull request closed without merging**